### DISCUSSION TITLE
Error on adding Filter

### DISCUSSION THREAD

**Discussion by Pratosh22** [ghiscoding/slickgrid-react#295] [Q&A] [answered]
On adding filter functionality , I'm getting this error 
<img width="752" alt="Screenshot 2024-02-08 at 8 48 32 PM" src="https://github.com/ghiscoding/slickgrid-react/assets/76246084/a403dbd3-3c1a-4c47-bf97-bf5482fb3095">
.On digging deep i found that the issue might be due to this value being undefined
<img width="1724" alt="Screenshot 2024-02-08 at 8 50 20 PM" src="https://github.com/ghiscoding/slickgrid-react/assets/76246084/ea2a7c6a-bb83-47c6-9731-92acd55e088d">

Any help is appreciated. Thanks :)

**Comment by ghiscoding** [comment]
I cannot help without a full reproduction (Stackblitz, GitHub repo or others...), all the Examples are working fine and they all have Cypress E2E tests running on every PR/Releases. So it has to be your code and again I cannot help just by looking at a print screen. The `this._cellContainerElm` is the div container of that filter, it should be filled, if you scroll up you can inspect the `init(args)` where `args` is where the filter div container is provided

**Reply by Pratosh22** [reply]
I have setup a sandbox with my implementation [here](https://codesandbox.io/p/github/Pratosh22/datagrid-poc/sandbox).

**Reply by ghiscoding** [reply]
Good thanks, I haven't tried it yet but quickly scrolling through the code and I see unrelated code. The highlighted code is not valid and doesn't belong there, these should be grid options only, they shouldn't be added in the slickgrid-react component itself, it's incorrect

![image](https://github.com/ghiscoding/slickgrid-react/assets/643976/2c7be6b2-3d1f-4629-ada7-4619dec8da06)

Your column has a `choices` property which is **not** a valid `Column` property, if you want to pass extra stuff then use the dedicated `params` which is generic and was created for that use case. You should use the `Column` interface from Slickgrid-React to highlight which properties are valid and invalid. Also the code above should probably be use a single ternary condition, I mean the last 2 conditions are both returning `FieldType.string` so... and also in case you didn't know you can use the `FieldType` or simply write `'string'`, they are both accepted.

![image](https://github.com/ghiscoding/slickgrid-react/assets/643976/e39a4458-c896-46c4-a5f8-6ca0be6cbd30)


You should also remove the `multiple-select-modified` dependency, I rewrote that lib and created a new project `multiple-select-vanilla` and started using it since slickgrid-react v2.x so you should not use the old ms-select since it might give you conflicts

![image](https://github.com/ghiscoding/slickgrid-react/assets/643976/11e1a8f9-fccc-4a37-a1a8-d30e2a575fcb)

**Reply by Pratosh22** [reply]
Thanks for the quick reply, I will make these changes!

**Reply by ghiscoding** [reply]
I don't see anything special in the code for it to throw, but I'm not a React developer. This project was created by another user and transferred back to me because I maintain all other repos, I'm not very string with React. Why don't you clone the [slickgrid-react-demos](https://github.com/ghiscoding/slickgrid-react-demos) and edit any of the example with your own code and see if that works? Because I mean all the examples are working fine and aren't throwing any errors, your sandbox also doesn't include any data and it doesn't run and I cannot reproduce the error from there. Also by the way, the text filter is the default filter, so if you mark your column as `filterable: true` and use `enableFiltering: true` then the default filter will be text input. 

In summary, like I said, I cannot help without a full repro. You did provide this sandbox but it doesn't start and has no data, so it's hard for me to do anything with it and it doesn't reproduce the error... For you to clone the demo repo and replace the data with your code seems like a better approach, you can remove the TypeScript stuff afterward but at least the demo repo is working and is updated every time I push a new release

**Reply by Pratosh22** [reply]
Ah,Thanks for the help! I will clone the demo and populate it with my data as you said. So far the library has been wonderful. Thank you :)

**Reply by Pratosh22** [reply, marked as answer]
Update: Fixed it! The issue was because i was assigning `id` of column as `question.id` which is something like 1000041, this made the function `getHeaderRowColumn` return undefined as this line `return headerRowTarget.children[idx]`  returns something like `children[100041]` which  does not exist in headerRowTarget. Using `index` as id fixed the issue

TL:DR: Changed column id to index to fix the issue.

**Reply by ghiscoding** [reply]
good to know you fixed it, you can also star ⭐ the project if you like it 😉
